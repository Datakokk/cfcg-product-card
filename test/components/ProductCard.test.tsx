import React from "react";
import renderer, { act } from "react-test-renderer";
import ProductCard from "../../src/components";
import { product1 } from "../data/products";

describe('Test on <ProductCard />', () => { 

    test('should display component correctly', () => { 

        const wrapper = renderer.create(
            <ProductCard product={ product1}>
                {
                    () => (
                        <h1>Product Card</h1>
                    )
                }
            </ProductCard>
        );
        expect ( wrapper.toJSON() ).toMatchSnapshot();
     });

     test('should increase counter', () => { 

        const wrapper = renderer.create(
            <ProductCard product={ product1}>
                {
                    ({ count, increaseBy }) => (
                        <>
                            <h2>Product Card</h2>
                            <span>{ count }</span>
                            <button onClick={ () => increaseBy(5)}></button>
                        </>
                    )
                }
            </ProductCard>
        );
        
        let tree = wrapper.toJSON();

        expect ( tree ).toMatchSnapshot();

        act(() => {
            (tree as any).children[2].props.onClick();
          });

        tree = wrapper.toJSON();

        expect((tree as any).children[1].children[0]).toBe("5")
     });
 })