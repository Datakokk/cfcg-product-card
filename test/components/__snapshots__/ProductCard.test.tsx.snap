// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <ProductCard /> should display component correctly 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;

exports[`Test on <ProductCard /> should increase counter 1`] = `
<div
  className="productCard undefined"
>
  <h2>
    Product Card
  </h2>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;
